File: FrontEnd/karuna/src/services/donorService.js
import axios from 'axios';

const baseURL = 'http://localhost:8080';


export const registerDonor = async (donor) => {
  try {
    const response = await axios.post(`${baseURL}/donor/register`, donor);
    return response.data;
  } catch (error) {
    throw error.response.data;
  }
};

export const loginDonor = async (loginDTO) => {
  try {
    const response = await axios.post(`${baseURL}/donor/login`, loginDTO);
    return response.data;
  } catch (error) {
    throw error.response.data;
  }
};

export const logoutDonor = async (donorId) => {
  try {
    const response = await axios.post(`${baseURL}/donor/logout/${donorId}`);
    return response.data;
  } catch (error) {
    throw error.response.data;
  }
};

export const updateAccount = async (donorId, donor) => {
  try {
    const response = await axios.put(`${baseURL}/donor/${donorId}`, donor);
    return response.data;
  } catch (error) {
    throw error.response.data;
  }
};

export const deleteAccount = async (donorId) => {
  try {
    const response = await axios.delete(`${baseURL}/donor/${donorId}`);
    return response.data;
  } catch (error) {
    throw error.response.data;
  }
};

export const donateToReceiver = async (receiverId, donorId, item) => {
  try {
    const response = await axios.post(
      `${baseURL}/donor/donate/receiver/${receiverId}/donor/${donorId}`,
      item
    );
    return response.data;
  } catch (error) {
    throw error.response.data;
  }
};

export const donate = async (donorId, item) => {
  try {
    const response = await axios.post(`${baseURL}/donor/donate/donor/${donorId}`, item);
    return response.data;
  } catch (error) {
    throw error.response.data;
  }
};

export const donateToCampaign = async (campaignId, donorId, item) => {
  try {
    const response = await axios.post(
      `${baseURL}/donor/donate/campaign/${campaignId}/donor/${donorId}`,
      item
    );
    return response.data;
  } catch (error) {
    throw error.response.data;
  }
};

export const viewHistory = async (donorId) => {
  try {
    const response = await axios.get(`${baseURL}/donor/history/${donorId}`);
    return response.data;
  } catch (error) {
    throw error.response.data;
  }
};

export const viewReceivers = async () => {
  try {
    const response = await axios.get(`${baseURL}/donor/receivers`);
    return response.data;
  } catch (error) {
    throw error.response.data;
  }
};

export const viewCampaigns = async () => {
  try {
    const response = await axios.get(`${baseURL}/donor/campaigns`);
    return response.data;
  } catch (error) {
    throw error.response.data;
  }
};

File: Karuna/src/main/java/com/app/karuna/service/DonorService.java
package com.app.karuna.service;

import java.util.List;

import com.app.karuna.dto.LoginDTO;
import com.app.karuna.entity.Campaign;
import com.app.karuna.entity.Donor;
import com.app.karuna.entity.Item;
import com.app.karuna.entity.Receiver;

public interface DonorService {
	
	public Donor registerDonor(Donor donor);
	
	public Donor loginDonor(LoginDTO loginDTO);
	
	public boolean logoutDonor(Long donorId);
	
	public Donor updateAcc(Donor donor);
	
	public boolean deleteAcc(Long donorId);
	
	public Item donateToReceiver(Long donorId,Long receiverId,Item item); 
	
	public Item donate(Long donorId,Item item);
	
	public Item donateToCampaign(Long donorId,Long campaignId,Item item);
	
	public List<Item> viewHistory(Long donorId);
	
	public List<Receiver> viewReceivers();

	public List<Campaign> viewCampaigns();
}
